src/utils/extract.js

```diff
@@ -0,0 +1,96 @@
+const htmlparser = require("htmlparser2")
+
+function parseIndentDescriptor(indentDescriptor) {
+  const match = /^(\+)?(tab|\d+)$/.exec(indentDescriptor)
+  if (!match) {
+    return { relative: false, spaces: "auto" }
+  }
+  return {
+    relative: match[1] === "+",
+    spaces: match[2] === "tab" ? "\t" : Array(Number(match[2]) + 1).join(" "),
+  }
+}
+
+function iterateStyles(code, onStyle) {
+  let index = 0
+  let currentStyle = null
+
+  const parser = new htmlparser.Parser({
+    onopentag(name) {
+      // Test if current tag is a valid <script> tag.
+      if (name !== "style") {
+        return
+      }
+      currentStyle = ""
+    },
+
+    onclosetag(name) {
+      if (name !== "style" || currentStyle === null) {
+        return
+      }
+      onStyle(code.slice(index, parser.startIndex - currentStyle.length), currentStyle)
+      index = parser.startIndex
+      currentStyle = null
+    },
+
+    ontext(data) {
+      if (currentStyle === null) {
+        return
+      }
+      currentStyle += data
+    },
+  })
+
+  parser.parseComplete(code)
+}
+
+export default function extract(code, rawIndentDescriptor) {
+  const indentDescriptor = parseIndentDescriptor(rawIndentDescriptor)
+  const map = []
+  let resultCode = ""
+  let lineNumber = 1
+
+  iterateStyles(code, function (previousCode, stylecode) {
+
+    // Mark that we're inside a <style> a tag and push all new lines
+    // in between the last </style> tag and this <style> tag to preserve
+    // location information.
+    const newLines = previousCode.match(/\r\n|\n|\r/g)
```

Comment:
We haven't worrying about the old mac format. I'd say leave it out here, to stick with the convention, and if you think that's a bad idea raise an issue so we fix it everywhere at once.